data-reporting-service/post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java

```diff
@@ -0,0 +1,150 @@
+
+    private final PatientRepository patientRepository;
+    private final ProviderRepository providerRepository;
+    private final OrganizationRepository organizationRepository;
+
+    @KafkaListener(topics = {"${spring.kafka.topic.investigation}"})
+    public void postProcessInvestigationMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                                @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.investigation-confirmation}"})
+    public void postProcessInvestigationConfirmationMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                                            @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.investigation-notification}"})
+    public void postProcessInvestigationNotificationMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                                            @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.organization}"})
+    public void postProcessOrganizationMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                               @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.patient}"})
+    public void postProcessPatientMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                          @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.provider}"})
+    public void postProcessProviderMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                          @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @Scheduled(fixedDelay = 5000)
```

Comment:
Is this interval likely to change?

Reply:
This can be changed based on our requirements